### PR TITLE
Disable stacktrace unit tests when Trilinos is enabled

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
Configuration on Ride now succeeds with automated Trilinos scripts.

**Comment by crtrott** [comment]
This eliminates the requirement for adding `-rdynamic`.

**Comment by ndellingwood** [comment]
Cherry-pick to develop branch #2683

**Comment by masterleinad** [comment]
What's the rationale here? Is it only just that one machine that can't handle `-rdynamic` or are there problems whenever `Kokkos` is built through `Trilinos`?

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -287,6 +287,7 @@ KOKKOS_ADD_TEST_EXECUTABLE(
 )
 # We need -rdynamic on GNU platforms for the stacktrace functionality
 # to work correctly with shared libraries
+if(NOT KOKKOS_HAS_TRILINOS)
 SET_PROPERTY(TARGET StackTraceTestExec PROPERTY ENABLE_EXPORTS 1)
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: The issue is that the exe target name with TriBITS.  You should have added a comment.

**Pull request merged**